server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -308,7 +334,9 @@ public InternalEngine(EngineConfig engineConfig) {
             }
             this.lastRefreshedCheckpointListener = new LastRefreshedCheckpointListener(localCheckpointTracker.getProcessedCheckpoint());
             this.internalReaderManager.addListener(lastRefreshedCheckpointListener);
-            maxSeqNoOfUpdatesOrDeletes = new AtomicLong(SequenceNumbers.max(localCheckpointTracker.getMaxSeqNo(), translog.getMaxSeqNo()));
+            maxSeqNoOfUpdatesOrDeletes = new AtomicLong(
+                SequenceNumbers.max(localCheckpointTracker.getMaxSeqNo(), translogManager.getTranslog().getMaxSeqNo())
```

Comment:
Not a big fan of this indirection. I think refactoring utility methods like `getMaxSeqNo` directly to the `TranslogManager` instead of invoking `getTranslog` everywhere is cleaner?

Reply:
Update: I just noticed the previous review comments. Looks like you're planning to decouple this further in follow up PRs. So disregard.